src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryApiServiceSpec.scala

```diff
@@ -370,3 +299,96 @@ class LibraryApiServiceSpec extends BaseServiceSpec with LibraryApiService with
     }
   }
 }
+
+/**
+  * This class represents a version of the application with Library APIs that can run a single request.
+  */
+class MockLibraryApiServiceApp(val request: HttpRequest) extends MockApplication with LibraryApiService {
+  val libraryServiceConstructor: (UserInfo) => LibraryService = LibraryService.constructor(app)
+  val ontologyServiceConstructor: () => OntologyService = OntologyService.constructor(app)
+  def checkRequest: (HttpResponse, StatusCode) = {
+    request ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {(response, status)}
+  }
+}
```

Comment:
each `MockLibraryApiServiceApp` reuses the same `app`, which means you're reusing the daos associated with that app. Is that really what you intended to do? It seems that you intended to create a new app/dao for each test.

Reply:
Update: I was wrong! Ignore this.